torch/onnx/utils.py

```diff
@@ -1653,17 +1654,17 @@ def _export_file(
             for k, v in export_map.items():
                 z.writestr(k, v)
     elif export_type == _exporter_states.ExportTypes.DIRECTORY:
-        if os.path.exists(f):
-            assert os.path.isdir(f)
+        if os.path.exists(f):  # type: ignore[arg-type]
```

Comment:
I think you can assert f is not `io.BytesIO`

Reply:
MYPY is not happy PathLike without specifying what's inside.

Reply:
Like

```python
    elif export_type == _exporter_states.ExportTypes.DIRECTORY:
        assert not isinstance(io.BytesIO, f)
        assert os.path.isdir(f)  # type: ignore[arg-type]
        if not os.path.exists(f):  # type: ignore[arg-type]
            os.makedirs(f)  # type: ignore[arg-type]
```
?

Reply:
I use raise instead, because I think these are from user? It doesn't make MYPY happy though. Every line is using os functions, it's going to complain about PathLike type. I would leave this as it is.

Reply:
hmm OK. I thought os.path.exists could take str as well. nvm